### PR TITLE
Waypoints rework

### PR DESCRIPTION
**Pull request by CommandrMoose**
Reworks the UI for the waypoints saving to be simplified. The save screen is now only available under the "WAYPOINTS" heading. 

image

### CODE REVIEW

File: common/src/main/java/whocraft/tardis_refined/client/screen/waypoints/WaypointListScreen.java [hunk 1]
```diff
@@ -27,18 +31,23 @@ public class WaypointListScreen extends SelectionScreen {
...
     private int leftPos, topPos;
 
+    private SpriteIconButton loadButton;
+    private SpriteIconButton editButton;
+    private SpriteIconButton trashButton;
+
     public static ResourceLocation MONITOR_TEXTURE = new ResourceLocation(TardisRefined.MODID, "textures/gui/monitor.png");
-    Collection<TardisNavLocation> WAYPOINTS = new ArrayList<>();
-    TardisNavLocation tardisNavLocation = TardisNavLocation.ORIGIN;
+    Collection<TardisWaypoint> WAYPOINTS = new ArrayList<>();
```
Comment (50ap5ud5) [inline common/src/main/java/whocraft/tardis_refined/client/screen/waypoints/WaypointListScreen.java]: Specify access modifier i.e. protected, private, public etc.

File: common/src/main/java/whocraft/tardis_refined/client/screen/waypoints/WaypointListScreen.java [hunk 2]
```diff
@@ -27,18 +31,23 @@ public class WaypointListScreen extends SelectionScreen {
...
+    TardisWaypoint waypoint = null;
```
Comment (50ap5ud5) [inline common/src/main/java/whocraft/tardis_refined/client/screen/waypoints/WaypointListScreen.java]: Specify access modifier i.e. protected, private, public etc.

File: common/src/main/java/whocraft/tardis_refined/client/screen/waypoints/WaypointListScreen.java [hunk 3]
```diff
@@ -115,15 +159,20 @@ public void renderBackground(GuiGraphics guiGraphics, int i, int j, float f) {
...
 
         guiGraphics.blit(MONITOR_TEXTURE, leftPos, topPos, 0, 0, imageWidth, imageHeight);
 
-        if(WAYPOINTS.isEmpty()) {
-            Font font = this.font;
-            guiGraphics.drawCenteredString(font, noWaypointsLabel, width / 2, height / 2 - 10, ChatFormatting.GOLD.getColor());
+        ScreenHelper.renderWidthScaledText(Component.translatable(ModMessages.UI_MONITOR_WAYPOINTS).getString(), guiGraphics, Minecraft.getInstance().font, width / 2 - 96, textOffset, Color.LIGHT_GRAY.getRGB(), textScale * 2, 1F, false);
+
+        if (WAYPOINTS.isEmpty()) {
+            ScreenHelper.renderWidthScaledText(noWaypointsLabel.getString(), guiGraphics, Minecraft.getInstance().font, width / 2 - 96, textOffset + 15, Color.LIGHT_GRAY.getRGB(), textScale * 2, 1F, false);
```
Comment (50ap5ud5) [inline common/src/main/java/whocraft/tardis_refined/client/screen/waypoints/WaypointListScreen.java:174]: Move field on Line 56 to top of class file for consistency. Add access modifier i.e. public, private, protected

``Line 56:  Component noWaypointsLabel = Component.translatable(ModMessages.UI_MONITOR_NO_WAYPOINTS);``

File: common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisWaypointManager.java [hunk 4]
```diff
@@ -4,41 +4,48 @@
...
-        waypointMap.remove(name);
+    public void editWaypoint(TardisWaypoint waypoint) {
+
+        waypoints.removeIf(x -> x.getId().equals(waypoint.getId()));
+        waypoints.add(waypoint);
+
+        waypoints.forEach(x -> {
+            System.out.println(x.getId());
+            System.out.println(x.getLocation().getName());
+        });
```
Comment (50ap5ud5) [inline common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisWaypointManager.java]: Remove prints

File: common/src/main/java/whocraft/tardis_refined/client/screen/waypoints/WaypointListScreen.java [hunk 5]
```diff
@@ -55,44 +64,79 @@ protected void init() {
...
+                if (waypoint != null) {
+                    new RemoveWaypointEntryMessage(waypoint.getId()).send();
                 }
             }
         });
 
 
-        addSubmitButton(width / 2 + 85, (height) / 2 - 35);
-        addTrashIcon(width / 2 + 85, (height) / 2 - 55);
+        SpriteIconButton spriteButtonNew = this.addRenderableWidget(CommonTRWidgets.imageButton(20, Component.translatable("Submit"), (arg) -> {
```
Comment (50ap5ud5) [inline common/src/main/java/whocraft/tardis_refined/client/screen/waypoints/WaypointListScreen.java]: Use a more meaningful variable name. Declare as a field if this is being reused like the load, edit button etc,

**Review by 50ap5ud5 — CHANGES_REQUESTED** [review]
Minor cleanup

File: common/src/main/java/whocraft/tardis_refined/client/screen/waypoints/WaypointListScreen.java (hunk 5 above)
Comment (CommandrMoose) [reply common/src/main/java/whocraft/tardis_refined/client/screen/waypoints/WaypointListScreen.java]: Addressed

File: common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisWaypointManager.java (hunk 4 above)
Comment (CommandrMoose) [reply common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisWaypointManager.java]: Addressed

File: common/src/main/java/whocraft/tardis_refined/client/screen/waypoints/WaypointListScreen.java (hunk 3 above)
Comment (CommandrMoose) [reply common/src/main/java/whocraft/tardis_refined/client/screen/waypoints/WaypointListScreen.java:174]: Addressed

File: common/src/main/java/whocraft/tardis_refined/client/screen/waypoints/WaypointListScreen.java (hunk 2 above)
Comment (CommandrMoose) [reply common/src/main/java/whocraft/tardis_refined/client/screen/waypoints/WaypointListScreen.java]: Addressed

File: common/src/main/java/whocraft/tardis_refined/client/screen/waypoints/WaypointListScreen.java (hunk 1 above)
Comment (CommandrMoose) [reply common/src/main/java/whocraft/tardis_refined/client/screen/waypoints/WaypointListScreen.java]: Addressed

**Review by 50ap5ud5 — APPROVED** [review]

**Pull request merged**